### PR TITLE
feat: implement last 4 steps of apple attestation

### PR DESCRIPTION
**Pull request by bryce-mcmath**
Pushing this PR so @jleach can try it out and give feedback before we continue with moving the relevant code from `apple.py main()` to `verify_attestation_statement()` and add caching for end to end functionality

### CODE REVIEW

File: src/apple.py [hunk 1]
```diff
@@ -206,15 +207,38 @@ def main():
...
-    # 7. Verify that the authenticator data’s counter field equals 0.
+    app_id = 'L796QSLV3E.ca.bc.gov.BCWallet'
+    app_id_bytes = app_id.encode('utf-8')
+    app_id_hash = hashlib.sha256(app_id_bytes).hexdigest()
+    auth_app_id_hash = apple_attestation_object['authData'][:32].hex()
+    print(auth_app_id_hash == app_id_hash)
+
+    # 7. Verify that the authenticator data’s counter field equals 0. See 
+    # https://www.w3.org/TR/webauthn/#sctn-attestation for byte start and end points.
+    counter_start = 33
```
Comment (jleach) [inline src/apple.py:220]: constants.ts?

**Review by jleach — CHANGES_REQUESTED** [review]

File: src/apple.py [hunk 2]
```diff
@@ -206,15 +207,38 @@ def main():
...
+    aaguid_start = 37
+    aaguid_end = 53
+    aaguid = apple_attestation_object['authData'][aaguid_start:aaguid_end]
+    print(aaguid == bytearray(b'appattestdevelop') or aaguid == bytearray(b'appattest\x00\x00\x00\x00\x00\x00\x00'))
+
     # 9. Verify that the authenticator data’s credentialId field is the same as the
     # key identifier.
+    key_identifier = bytes_value
+    cred_id_length = len(key_identifier)
+    cred_id_start = 55
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply src/apple.py]: done

File: src/apple.py (hunk 1 above)
Comment (bryce-mcmath) [reply src/apple.py]: done

File: src/apple.py [hunk 3]
```diff
@@ -206,15 +207,38 @@ def main():
...
     print('pub key match =', hash == hash_hex)
 
-
-if __name__ == "__main__":
-    main()
-
     # 6. Compute the SHA256 hash of your app’s App ID, and verify that it’s the same as the
     # authenticator data’s RP ID hash.
-    # 7. Verify that the authenticator data’s counter field equals 0.
+    app_id = 'L796QSLV3E.ca.bc.gov.BCWallet'
```
[the remark this answers is not in the document]
Comment (bryce-mcmath) [reply src/apple.py]: done

File: src/apple.py [hunk 4]
```diff
@@ -131,90 +134,114 @@ def create_hash_from_pub_key(cred_certificate):
...
-    # data buffer in the array (credcert). Verify the validity of the certificates using
-    # Apple’s App Attest root certificate.
-
-    verify_x5c_certificates(apple_attestation_object)
-
-    # 2. Create clientDataHash as the SHA256 hash of the one-time challenge your server sends
-    # to your app before performing the attestation, and append that hash to the end of the
-    # authenticator data (authData from the decoded object).
+def create_app_id_hash():
+    app_id = 'L796QSLV3E.ca.bc.gov.BCWallet'
```
Comment (jleach) [inline src/apple.py:138]: To constants.py?

**Review by jleach — CHANGES_REQUESTED** [review]

File: src/apple.py (hunk 4 above)
Comment (bryce-mcmath) [reply src/apple.py]: done

**Review by jleach — APPROVED** [review]

**Pull request merged**